src/components/AddressBlocked.tsx

```diff
@@ -0,0 +1,15 @@
+import { ReactNode } from 'react';
+import { useAddressAllowed } from 'src/hooks/useAddressAllowed';
+import { useWeb3Context } from 'src/libs/hooks/useWeb3Context';
+import { AddressBlockedModal } from './AddressBlockedModal';
+
+export const AddressBlocked = ({ children }: { children: ReactNode }) => {
+  const { currentAccount } = useWeb3Context();
+  const { isAllowed } = useAddressAllowed();
+
+  if (!isAllowed) {
+    return <AddressBlockedModal address={currentAccount} />;
+  }
+
+  return <>{children}</>;
+};
```

Comment:
🔥 Dope, works for every page

One thing we could do, this adds the navbar to the top:

```jsx
<MainLayout>
  <AddressBlockedModal address={currentAccount} />;
</MainLayout>
```

Reply:
Nice, good call. Looks better with the nav bar